### ISSUE TITLE
AnnouncerTest sporadically fails in master in TravisCI

### ISSUE THREAD

**Issue by drcrallen** [druid-io/druid#1393]
AnnouncerTest seems to be failing pretty regularly in TravisCI these days.

Specifically `AnnouncerTest.testSessionKilled` will often fail when checking the data in ZK

**Comment by drcrallen** [comment]
I looked into this a bit, and it might stem from assumptions of atomicity in the announcer. This was partially addressed in https://github.com/druid-io/druid/pull/1109 but not completely.

Curator does provide a `org.apache.curator.framework.recipes.nodes.PersistentEphemeralNode` which accomplishes exactly what the announcer intends to accomplish, but has two major distinctions:

1. There is no atomicity (all tasks are background tasks)
1. There is no compression like is currently enforced in the Announcer.

The Announcer probably needs re-written using the mechanism in PersistentEphemeralNode or similar.

Additionally, any place which uses the announcer probably needs assumptions of atomicity removed.

**Comment by himanshug** [comment]
this is duplicate of #2167

**Comment by drcrallen** [comment]
wouldn't 2167 be a duplicate of this one?

**Comment by himanshug** [comment]
@drcrallen ok, let me rephrase, "this is duplicated by 2167" :)
